### PR TITLE
Sign in with Ethereum: re-enable warning UI for mismatched domains / disable domain binding

### LINKED ISSUE #17707: [Bug]: Sign In with Ethereum popup terminated unexpectedly
### Describe the bug

Sign In with Ethereum popup terminated unexpectedly 

https://user-images.githubusercontent.com/53189696/218042372-ef68235e-51a1-402d-a696-ec8716c5d7f5.mov

### Steps to reproduce

1. Login to the extension
2. Navigate to the test dapp
3. Click Sign In With Ethereum (Bad Account)
4. Check the background console

### Error messages or log output

```shell
SES_UNHANDLED_REJECTION: Error: SIWE domain is not valid: "metamask.github.io" !== "metamask.badactor.io"
  at y.addUnapprovedMessage (personal-message-manager.js:159:15)
  at personal-message-manager.js:101:12
  at new Promise (<anonymous>)
  at y.addUnapprovedMessageAsync (personal-message-manager.js:94:12)
  at De.newUnsignedPersonalMessage (metamask-controller.js:2790:48)
  at wallet.js:161:1
  at async createAsyncMiddleware.js:45:1
```

### Version

10.25.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

Fixes https://github.com/MetaMask/metamask-extension/issues/17707

Hotfix targeting v10.26.2

- unblocks mismatched domain support
- reverts logic from #16616
- we may re-add error handling here with proper error handling #18184
- temp fix SIWE actionable-message styles
  - Improved style fix in follow-up PR since it may effect other <ActionableMessage> instances: https://github.com/MetaMask/metamask-extension/pull/18207

### Before

https://user-images.githubusercontent.com/53189696/218042372-ef68235e-51a1-402d-a696-ec8716c5d7f5.mov

previous warning UI:
Screen Shot 2023-03-16 at 10 22 22 AM

### After
Screen Shot 2023-03-16 at 10 22 29 AM
Screen Shot 2023-03-16 at 10 22 13 AM
Screen Shot 2023-03-16 at 10 38 04 AM

## Manual Testing Steps

1. Login to the extension
2. Navigate to the test dapp
3. Click Sign In With Ethereum (Bad Account)
4. Observe warning flow 

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [x] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by skgbafa** [comment]
The code change here looks solid. I think the components used in the warning was modified in other PRs, thus the odd views. @digiwand There is no other code to worry about from a functionality perspective

**Comment by digiwand** [comment]
Thanks @skgbafa! for now, I've added code to fix the warning UI. We can follow-up with a proper fix and fine-tune the UI after

**Review by jpuri — DISMISSED** [review]

File: app/scripts/lib/personal-message-manager.test.js [hunk 1]
```diff
@@ -148,45 +148,4 @@ describe('Personal Message Manager', () => {
...
-      const result = messageManager.getMsg(msgId);
-      expect(result.msgParams.siwe.isSIWEMessage).toStrictEqual(true);
-      // non-siwe message
-      const msgId2 = await messageManager.addUnapprovedMessage(
-        nonSiweMsgParams,
-        request,
-      );
-      const result2 = messageManager.getMsg(msgId2);
-      expect(result2.msgParams.siwe.isSIWEMessage).toStrictEqual(false);
-    });
```
Comment (jpuri) [inline app/scripts/lib/personal-message-manager.test.js:180]: May be above test case still make sense.

**Review by danjm — APPROVED** [review]
LGTM

File: ui/components/app/signature-request-siwe/index.scss [hunk 2]
```diff
@@ -21,18 +21,25 @@
     box-shadow: 0 0 7px 0 rgba(0, 0, 0, 0.08);
   }
 
+  /** @todo replace ActionableMessage or remove overwritten code. */
```
Comment (digiwand) [inline ui/components/app/signature-request-siwe/index.scss:24]: this is addressed in this follow-up PR: https://github.com/MetaMask/metamask-extension/pull/18207

**Review by jpuri — APPROVED** [review]

**Comment by naugtur** [comment]
Follow-up to this 
https://github.com/MetaMask/metamask-extension/issues/18188#issuecomment-1497120809

**Pull request merged**